nixos/modules/virtualisation/nixos-containers.nix

```
@@ -272,7 +272,17 @@ let
   };
 
   system = config.nixpkgs.localSystem.system;
-  kernelVersion = config.boot.kernelPackages.kernel.version;
+
+  # Sometimes ^1 the `kernel.version` property is not present; when it happens,
+  # we want for all kernel-based assertions to pass and, in the worst case,
+  # error-out at runtime.
+  #
+  # [1]: https://github.com/NixOS/nixpkgs/issues/38509#issuecomment-817200996
+  kernelVersion =
+    if lib.hasAttrByPath ["boot" "kernelPackages" "kernel" "version"] config then
+      config.boot.kernelPackages.kernel.version
+    else
+      null;
```

Perhaps I'm missing why this wouldn't work, but couldn't you just say 
```suggestion
  kernelVersion = config.boot.kernelPackages.kernel.version or null;
```

---

hah, I was under the impression that `or` applies to the last element only (i.e. this would fail if `kernelPackages` was empty); thanks!

---

Yeah, I also had to check, but it seems to be smart enough :)